src/android/Capture.java

```diff
@@ -393,25 +436,19 @@ public void onImageActivityResult(Request req) {
     }
 
     public void onVideoActivityResult(Request req, Intent intent) {
-        Uri data = null;
 
-        if (intent != null){
-            // Get the uri of the video clip
-            data = intent.getData();
-        }
-
-        if( data == null){
+        if(videoUri == null){
             File movie = new File(getTempDirectoryPath(), "Capture.avi");
-            data = Uri.fromFile(movie);
+            videoUri = Uri.fromFile(movie);
         }
 
         // create a file object from the uri
-        if(data == null) {
+        if (videoUri == null) {
             pendingRequests.resolveWithFailure(req, createErrorObject(CAPTURE_NO_MEDIA_FILES, "Error: data is null"));
         }
         else {
-            req.results.put(createMediaFile(data));
-
+            req.results.put(createMediaFile(videoUri));
+            videoUri = null;
```

Comment:
Does this need to be in a `finally` type block? I can believe something happening above this that then never frees up videoUri

Reply:
That's conceivable